### PR TITLE
04497 token pause unpause prehandle

### LINKED ISSUE #4497: Implement `TokenPauseUnpause` pre handle

### PR DESCRIPTION
**Pull request by georg-getz**
This PR adds the prehandle logic for token pause/unpause

**Related issue(s)**:

Fixes #4497 

SonarCloud bug isn't real

### CODE REVIEW

File: hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/TokenPreTransactionHandlerImpl.java [hunk 1]
```diff
@@ -172,14 +173,47 @@ public TransactionMetadata preHandleUpdateTokenFeeSchedule(
...
     public TransactionMetadata preHandleUnpauseToken(
             @NonNull final TransactionBody txn, @NonNull final AccountID payer) {
         Objects.requireNonNull(txn);
-        throw new NotImplementedException();
+        final var op = txn.getTokenUnpause();
+        final var meta =
+                new SigTransactionMetadataBuilder(accountStore).payerKeyFor(payer).txnBody(txn);
+
+        if (op.hasToken()) {
+            addPauseKey(op.getToken(), meta);
```
Comment (rbair23) [inline hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/TokenPreTransactionHandlerImpl.java:199]: It seems that calling `addPauseKey` for an "unpause" handler is weird. Is this expected?

File: hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/TokenPreTransactionHandlerImpl.java [hunk 2]
```diff
@@ -172,14 +173,47 @@ public TransactionMetadata preHandleUpdateTokenFeeSchedule(
...
+        } else {
+            meta.status(ResponseCodeEnum.INVALID_TOKEN_ID);
+        }
+
+        return meta.build();
+    }
+
+    private void addPauseKey(TokenID tokenId, SigTransactionMetadataBuilder meta) {
```
Comment (rbair23) [inline hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/TokenPreTransactionHandlerImpl.java:207]: Some javadoc would be nice

File: hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/TokenPreTransactionHandlerImpl.java (hunk 1 above)
Comment (Neeharika-Sompalli) [reply hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/TokenPreTransactionHandlerImpl.java:199]: yeah, token's pause key is used to sign for `tokenPause` and `tokenUnpause`.

File: hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/TokenPreTransactionHandlerImpl.java [hunk 3]
```diff
@@ -172,14 +173,47 @@ public TransactionMetadata preHandleUpdateTokenFeeSchedule(
     public TransactionMetadata preHandlePauseToken(
             @NonNull final TransactionBody txn, @NonNull final AccountID payer) {
         Objects.requireNonNull(txn);
-        throw new NotImplementedException();
+        final var op = txn.getTokenPause();
+        final var meta =
+                new SigTransactionMetadataBuilder(accountStore).payerKeyFor(payer).txnBody(txn);
```
Comment (Neeharika-Sompalli) [inline hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/TokenPreTransactionHandlerImpl.java:178]: We might pass in `SigTransactionMetadataBuilder` into the methods in future changes.

**Review by Neeharika-Sompalli — APPROVED** [review]
LGTM !

**Review by stoqnkpL — APPROVED** [review]

**Pull request merged**